common/state/state_kv.cpp

```diff
@@ -604,12 +575,13 @@ void pstate::trie_node::do_read_value(
     dn_io.read_across_data_nodes(bo, vs, value);
 }
 
-void pstate::trie_node::do_delete_value(data_node_io& dn_io, trie_node_header_t* header)
+void pstate::trie_node::do_delete_value(data_node_io& dn_io, trie_node& node)
 {
-    pdo::error::ThrowIf<pdo::error::RuntimeError>(
-        !header->hasChild, "delete value, header must have child");
-    block_offset current_child_bo;
-    current_child_bo.deserialize_offset(*goto_child_offset(header));
+    block_offset current_child_bo(node.node.child_offset);
+    //current_child_bo.deserialize_offset(*goto_child_offset(header));
+
+    //TODO MODIFY DELETE TO WRITE A DELETED VALUE HEADER (1 byte)
+    //TODO remove bytearray from from delete value in data node
 
```

Comment:
remove all TODOs before you commit. these can stay in your own repo, but don't commit them.

Reply:
You are going to make me crazy.